inference/cluster_store.py

```diff
@@ -82,8 +83,15 @@ def build_optimal_agglomerative_clusters(self):
     # Function to run agglomerative clustering 
     # with specified number of cluster: c -> int
     def build_agglomerative_clusters(self, c):
-        # TODO
-        pass
+        self.total_cluster = c
+        X = np.asarray(self.original_data)
+        clustering = AgglomerativeClustering(n_clusters=c).fit(X)
+        self.clusters = {}
+        for i in range(self.total_cluster):
+            self.clusters[i] = []
+        for i in range(len(clustering.labels_)):
```

Comment:
These labels should be referring to the corresponding label in `self.entity_map`.

```python
# The key in the map corresponds to the index of the feature array in data.

original_data = [
    [1,...],
    [26,...],
    [12,...]
]

entity_map = {
    0 : "entity1",
    1 : "entity2",
    2 : "entity3"
}
```